draft-packages/empty-state/KaizenDraft/EmptyState/EmptyState.tsx

```diff
@@ -96,9 +98,9 @@ export const EmptyState: React.VFC<EmptyStateProps> = ({
       <div className={styles.textSide}>
         <div className={styles.textSideInner}>
           <Heading
-            variant="heading-3"
+            variant={headingVariant ? headingVariant : "heading-3"}
+            tag={headingVariant ? undefined : "div"}
```

Comment:
I think we should probably create a `headingTag` prop as well, so that semantics and styling aren't tightly coupled. From my experience it's more rare that the styling and semantics actually align.

Reply:
Actually on second thought, that will mean you end up with 3 separate props for headings things, which is a bit gross.

Couple of options:
1. We expose the heading props directly in a single prop. Something like
```
<EmptyState
  ...
  heading={{
    children: "My heading",
    variant: "heading-3",
    tag: "h4"
  }}
```
We'd need to deprecate the `headingText` prop, and also set defaults for the `variant` and `tag` parts, which might be a bit tricky.

2. We add a render prop instead, which gives people ultimate flexibility to add whatever they want here.

I'm leaning towards option 2 because it's simple and gives people more flexibility. Interested to get @HeartSquared's take on it as the interim tech lead. If we were doing this from scratch I think it would be a different API completely, but these 2 options might be the most realistic given what we're already working with.

Reply:
I'd also suggest that either way we go with this, we should also have a couple of tests to test the new prop/s. If we're going the route of render we'd probably want a story example too 🤔

Reply:
I've personally been going down the pattern from option 1 for stylistically opinionated subcomponents. It is easier for the consumer to create consistency without the need for an extra import.

That being said, I'm not against option 2 if people think `heading={<Heading />}` is nicer to consume.

Both options are a breaking change regardless. As @dougmacknz said, if we were to rebuild the component, we would likely choose a different option altogether.

Reply:
The render prop could just be an additional prop, rather than replacing `headingText`, so that there's a simple option and a flexible option. That's what we've generally done when adding render props.

Reply:
This conversation reminded me of something I was going to take a look at next - the [Inline Notification component ](https://cultureamp.design/storybook/?path=/story/components-notification-inline-notification--default-kaizen-demo)heading is a H6, which causes a violation when used on our [Confidentiality config page](https://frontend-assets.usw2.dev-us.cultureamp-cdn.com/dev/survey-configuration/storybook/krypton/caitlin/accessibility-fixes--confidentiality-page/index.html?path=/story/confidentiality-confidentialitypage--default-view) as the heading levels skip from H3 to H6. It would be good if we could also have control over the heading level in that component, or use a Div to avoid the violation.

Reply:
Hey @caitlinmwillington we talked about this as team at standup and think the best course of action is to go with something closer to option 1 dug recommended originally, which should give granular enough control of the semantic without you needing to pull in the Heading as in as an import into your page. Super rough example but something along the lines of this:

```ts
export interface EmptyStateProps
  extends OverrideClassName<HTMLAttributes<HTMLDivElement>>,
    Pick<AnimatedSceneProps, "isAnimated" | "loop"> {
  ...
  // we could make this optional and give a deprecation warning on the prop
  /**
   * **Deprecated:** Use headingProps
   * @deprecated
   */
  headingText?: string | React.ReactNode
  // We could import Heading Props from "@kaizen/typography"
  headingProps?: HeadingProps
  ...
}
```

and then providing some defaults in the component restructuring that can be used in the component

```tsx
  export const EmptyState: React.VFC<EmptyStateProps> = ({
    ...
    headingText,
    headingProps = {
      variant: "heading-3",
      tag: "div",
      color: "dark"
    },
    ...
    
    return (
    ...
    <Heading
      variant={headingProps.variant}
      classNameOverride={styles.heading}
      tag={headingProps.tag}
      color={headingProps.color}
    >
      {headingProps.children || headingText}
    </Heading>
  ...
  )
```

Reply:
It might be good to pair on this one if you've got some time free this week to shorten the feedback loop? :)

Reply:
Thanks @mcwinter07 , sounds like a great idea! I have booked some time in for Thurs but feel free to move it if the time doesn't suit

Reply:
@mcwinter07 I have pushed those changes, added a test and updated the storybook to use the new heading props over the deprecated one

Reply:
Brill, just reviewing it now